pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -87,7 +87,33 @@ let
       args = rattrs (args // { inherit finalPackage overrideAttrs; });
       #              ^^^^
 
-      overrideAttrs = f0: makeDerivationExtensible (lib.extends (lib.toExtension f0) rattrs);
+      overrideAttrs =
+        f0:
+        let
+          extension = lib.toExtension f0;
+          prev = rattrs args;
+          thisOverlay = extension args prev;
+          # warnForBadVersionOverride = (thisOverlay ? version && !(thisOverlay ? src));
```

Comment:
When the predicate is this, a noop `overrideAttrs` (e.g. `hello.overrideAttrs (_: {}))`) still gets ~178 thunks added. Why?